test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/CharacterTest.scala

```diff
@@ -383,6 +383,159 @@ class CharacterTest {
     assertFalse(Character.isDigit('\uFBFC'))
   }
 
+  @Test def toLowerCase_Char(): Unit = {
+    // Low
+    assertEquals('\n', Character.toLowerCase('\n'))
+    // Ascii
+    assertEquals('0', Character.toLowerCase('0'))
+    assertEquals('a', Character.toLowerCase('a'))
+    assertEquals('a', Character.toLowerCase('A'))
+    assertEquals('f', Character.toLowerCase('F'))
+    assertEquals('z', Character.toLowerCase('Z'))
+    // compat/ucd
```

Comment:
These comments don't mean anything to me. What do they refer to?

Consider using comments that actually explain why the tests are interesting. I could see a few reasons:

* ASCII repertoire, not a letter (no change)
* ASCII repertoire, lowercase letter (no change)
* ASCII repertoire, uppercase letter
* Somewhere else in the BMP, not a letter (no change)
* Somewhere else in the BMP, lowercase letter (no change)
* Somewhere else in the BMP, titlecase letter
* Somewhere else in the BMP, uppercase letter
* Surrogate Chars (no change)

Reply:
Given the new implementations I suggest above, we should definitely have exhaustive tests for all the special characters in `toUpperCase` and `toTitleCase`.

We could also have 3 generic sanity tests for the overloads with Chars, of the form:
```scala
for (ch <- Character.MIN_VALUE to Character.MAX_VALUE)
  assertEquals(Character.toLowerCase(ch), Character.toLowerCase(ch.toInt).toChar)
```
(with the same for `toUpperCase` and `toTitleCase`). When we run those on the JVM, we make sure that the naive implementations we have for the Char overloads is indeed correct.